provider/cldr/src/transform/time_zones/convert.rs

```diff
@@ -111,266 +111,167 @@ impl From<TimeZoneNames> for ExemplarCitiesV1<'_> {
-                            !names.is_empty()
-                        },
-                    )
-                    .collect(),
-            },
-            overrides: other
-                .zone
-                .0
-                .into_tuple_vec()
-                .into_iter()
-                .flat_map(|(key, region)| {
-                    region.0.into_tuple_vec().into_iter().flat_map(
-                        move |(inner_key, place_or_region)| {
-                            let mut key = key.clone();
-                            key.push('/');
-                            key.push_str(&inner_key);
-                            match place_or_region {
-                                LocationOrSubRegion::Location(place) => vec![place]
-                                    .into_iter()
-                                    .filter_map(|inner_place| {
-                                        inner_place
-                                            .short_metazone_names()
-                                            .map(|format| (key.clone().into(), format.into()))
-                                    })
-                                    .collect::<Vec<_>>(),
-                                LocationOrSubRegion::SubRegion(region) => region
-                                    .into_tuple_vec()
-                                    .into_iter()
-                                    .filter_map(|(inner_key, place)| {
-                                        let mut key = key.clone();
-                                        key.push('/');
-                                        key.push_str(&inner_key);
-                                        place
-                                            .short_metazone_names()
-                                            .map(|format| (key.into(), format.into()))
-                                    })
-                                    .collect::<Vec<_>>(),
-                            }
-                        },
-                    )
-                })
-                .collect(),
-        }
-    }
-}
+long_short_impls!(
+    MetaZoneGenericNamesShortV1<'_>,
+    MetaZoneSpecificNamesShortV1<'_>,
+    short,
+    short_metazone_names
+);
 
-impl From<ZoneFormat> for MetaZoneSpecificNamesV1<'_> {
-    fn from(other: ZoneFormat) -> Self {
-        Self(
-            other
-                .0
-                .into_tuple_vec()
-                .into_iter()
-                .filter(|(key, _)| !key.eq("generic"))
-                .map(|(key, value)| {
-                    (
-                        Cow::Owned(
-                            key.parse::<TinyStr8>()
-                                .expect("Time-zone variant was not compatible with TinyStr8"),
-                        ),
-                        value.into(),
-                    )
-                })
-                .collect(),
-        )
-    }
+fn fold_format_into_zero_map_2d(
```

Comment:
Suggestion: Implement FromIterator on ZeroMap2d (building from a tuple vec of triples).